core/app/models/spree/store_credit.rb

```diff
@@ -23,10 +23,12 @@ class Spree::StoreCredit < Spree::PaymentSource
   validates_numericality_of :amount_used, { greater_than_or_equal_to: 0 }
   validate :amount_used_less_than_or_equal_to_amount
   validate :amount_authorized_less_than_or_equal_to_amount
+  validates :expires_at, presence: true, unless: ->(credit) { credit.category.non_expiring? }
```

Comment:
That makes some sense, but are we sure that everything in an `expiring` category _must_ be expiring? Or does the category mean it _may_ be expiring